### DISCUSSION TITLE
20.2.2 (critical bugfix on the 20.2.x LTS series)

### DISCUSSION THREAD

**Discussion by oesteban** [nipreps/fmriprep#2452] [General]
## Release Notes
Hot-fix release in the 20.2.x LTS series.
Addressing a bug on *fMRIPrep*'s resampling to standard spaces (#2444).
In addition, some other relevant bug fixes and performance improvements are built into this patch.

## CHANGES
* FIX: Feed *NiTransforms* with LTAs of type RAS2RAS (#2444)
* FIX: Add some clarity to ``BrokenProcessPool`` failures (#2436)
* FIX: Fall-back to initializing workflow in main process (#2435)
* FIX: Raise informative error when duplicate subworkflows are added (#2434)
* FIX: Non-existing path or JSON syntax error for ``--bids-filter-file`` should raise on error (#2331)
* FIX: Ignore SBRef files if ``--ignore sbref`` is passed (#2370)
* ENH: Relax requirement for *PyBIDS* databases to exist (#2429)
* ENH: Improve ``_get_series_len`` performance (#2406)
* ENH: Set and track NumPy's random seed (#2400)
* DOC: Skull-stripping is forced by default (#2430)
* MAINT: Pin ``nilearn==0.6.2`` (#2427)
* MAINT: Pin ``tedana==0.0.9a1`` for LTS branch (#2403)
* MAINT: Failing CI (#2401)

<hr /><em>This discussion was created from the release <a href='https://github.com/nipreps/fmriprep/releases/tag/20.2.2'>20.2.2 (critical bugfix on the 20.2.x LTS series)</a>.</em>

**Comment by oesteban** [comment]
@nipreps/fmriprep-contributors The latest LTS patch release is out! Please update if you were using 20.2.x series as it resolves a major bug when resampling data into standard space (#2444).

The Pypi package has been posted - https://pypi.org/project/fmriprep/20.2.2/ and the docker image should be available soon through the Docker hub.

**Reply by gdevenyi** [reply]
> In addition, some other relevant bug fixes and performance improvements are built into this patch.

Its too late now, but was this an appropriate choice to roll-up with a hotfix?

**Reply by oesteban** [reply]
Sorry, I responded on my phone and GH didn't thread correctly:

> Probably not, but given the resources we have (circleci free tier and developer time), the nature of the other changes, and the fact that we haven't released since November, this is the best we can do.
>
> Happy to talk more with the rest of the community involved

I will add one thing - the hotfix is really to 20.1.4 (which is pending for release after sorting out many little pitfalls of our release system, which was not exercised for a long while). The delta between 20.1.3 and 20.1.4 is just the hotfix.

So one of the changes (with admittedly the larger impact) of this release is porting the fix forward.

**Reply by gdevenyi** [reply]
> I will add one thing - the hotfix is really to 20.1.4 (which is pending for release after sorting out many little pitfalls of our release system, which was not exercised for a long while). The delta between 20.1.3 and 20.1.4 is just the hotfix.

That makes more sense then :+1:

**Comment by julfou81** [comment]
Hi, is it possible to build a singularity image for this new release? I don't manage right now.
I got this error message with the ```singularity build``` command
```ERROR MANIFEST_UNKNOWN: manifest unknown```
Thank you for this release!

**Reply by julfou81** [reply]
ok it works now, thanks!

**Reply by oesteban** [reply]
Hi @julfou81, the docker image has been posted very very recently. Can you give it a second try and post an issue on the repo if the problem persists?

**Reply by julfou81** [reply]
it works! Good job! Thanks!

**Reply by effigies** [reply]
@julfou81 Please note that the 20.2.2 image on Docker Hub is actually 20.1.4. We're looking into what went wrong with the upload.

**Reply by oesteban** [reply]
Hey @julfou81, yesterday I overwrote the 20.2.2 tag on Docker Hub. Please make sure you drop the old image and pull the new one.

```
docker image rm nipreps/fmriprep:20.2.2
docker pull nipreps/fmriprep:20.2.2
```

After that, please make sure to run:

```
docker run -it --rm nipreps/fmriprep:20.2.2 --version
```

you should get

```
fMRIPrep v20.2.2
```

**Reply by julfou81** [reply]
Hi @oesteban , thank you for letting me know! I follow your advice and it seems ok now for the singularity image:
 I did: 
 - ```singularity build fmriprep-20.2.2.simg docker://nipreps/fmriprep:20.2.2```
 - ```singularity run fmriprep-20.2.2.simg --version```
and got: ```fMRIPrep v20.2.2```

**Comment by smeisler** [comment]
I built the container from dockerhub (`singularity build fmriprep_20_2_2.simg docker://nipreps/fmriprep:20.2.2`), and the version is listed at 20.1.4 when I run `fmriprep --version`.

**Reply by effigies** [reply]
Yes, it is 20.1.4, for some reason. We're looking into what went wrong with the upload process.

**Reply by oesteban** [reply]
The pypi package also reports some 20.1.3+3 version. This is bizarre.

My git log:

```
commit 7b2203ffaa7be3d2e52e916020b0abca36b5482a (HEAD -> rel/20.2.2, tag: 20.2.2, upstream/maint/20.2.x, maint/20.2.x)
Author: Oscar Esteban <code@oscaresteban.es>
Date:   Fri Jul 16 11:39:42 2021 +0200

    rel(20.2.2): Update flagged versions [skip ci]

commit 2e7ea06cecbddeec0375ed11e39ea92d0568e34e
Author: Oscar Esteban <code@oscaresteban.es>
Date:   Fri Jul 16 11:38:38 2021 +0200

    rel(20.2.2): Update CHANGES [skip ci]

commit f4817e28e9e0f6aa617aa821a948f4a28bc6a24a
Merge: d70c1347 4ee2be4e
Author: Oscar Esteban <code@oscaresteban.es>
Date:   Fri Jul 16 11:31:48 2021 +0200

    Merge branch 'maint/20.1.x' into maint/20.2.x

commit 4ee2be4ed40d4c5c65ef37756edd70500390c61f
Author: Oscar Esteban <code@oscaresteban.es>
Date:   Fri Jul 16 11:22:50 2021 +0200

    rel(20.1.4): Update flagged versions [skip ci]

commit 1c315b9de9a582c49b70c8ff831d54f68b4d99c3
Author: Oscar Esteban <code@oscaresteban.es>
Date:   Fri Jul 16 11:18:56 2021 +0200

    rel(20.1.4): Update CHANGES [skip ci]
```

and supposedly 7b2203ffaa7be3d2e52e916020b0abca36b5482a is the commit sent to Circle...

**Reply by oesteban** [reply]
Okay, so the job deploying to docker hub loaded its cache from job 1421 (which happens to be a build of 20.1.4). So that's where things got confused for docker.

However, it seems that also the python package is wrong?

**Reply by oesteban** [reply]
Okay the python package should be fine, in principle - https://files.pythonhosted.org/packages/a9/59/316c4afaa0bb6a954dcac7f75ee2b1ed1686ec59f78a8a3b91486c722891/fmriprep-20.2.2.tar.gz

**Reply by oesteban** [reply]
I have manually pushed a new 20.2.2 image. Should be fine.

**Reply by oesteban** [reply]
https://github.com/nipreps/fmriprep/discussions/2452#discussioncomment-1015595

**Comment by bbfrederick** [comment]
Did you install fmriprep locally before sending to GitHub?  With rapidtide, I discovered that some version fields don't update until I deploy things locally (so my docker and pypi versions were always wrong if I didn't do that before syncing everything). But that may just be a versioneer quirk.

**Reply by oesteban** [reply]
Should be good now, we identified the cause (our circle cache messes up when more than one tag are being built in parallel, #2454). In this case 20.1.4 and 20.2.2 jobs were running interleaved.

I have manually uploaded the proper container.

Thanks for the suggestion!

**Comment by effigies** [comment]
Superseded by https://github.com/nipreps/fmriprep/discussions/3331
